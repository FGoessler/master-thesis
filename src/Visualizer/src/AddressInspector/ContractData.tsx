import * as React from "react";
import { ContractValue } from "../AddressInspector/AddressInfoService";

export default function ContractData(props: { contractData: ContractValue[] | undefined }) {

  if (!props.contractData || (props.contractData && props.contractData[0].value === "0x")) {
    return (<div>No Contract Data</div>);
  }

  return (
    <div>
      <h4>Contract Data</h4>
      <table>
        <tbody>
        {props.contractData.map((c: ContractValue) =>
          <tr key={c.label}>
            <td>{c.label}</td>
            <td className="hash">{isTokenId(c) ? "0x" + c.value.toString(16) : c.value.toString(10)}</td>
          </tr>
        )}
        </tbody>
      </table>
    </div>
  );
}

function isTokenId(contractValue: ContractValue) {
  return contractValue.label === "activeTokenId" || contractValue.label === "soonExpiringTokenId";
}
