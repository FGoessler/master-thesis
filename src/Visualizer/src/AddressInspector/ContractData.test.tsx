import * as React from "react";
import * as ReactDOM from "react-dom";
import ContractData from "./ContractData";

it("lists the contract data", async () => {
  const data = [
    {label: "a", value: 1},
    {label: "b", value: "123"},
  ];
  const item = document.createElement("div");
  ReactDOM.render(
    <ContractData
      contractData={data}
    />,
    item);
  expect(item).toMatchSnapshot();
});
