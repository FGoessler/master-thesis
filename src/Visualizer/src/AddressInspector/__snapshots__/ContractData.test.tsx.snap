// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lists the contract data 1`] = `
<div>
  <div
    data-reactroot=""
  >
    <h4>
      Contract Data
    </h4>
    <table>
      <tbody>
        <tr>
          <td>
            a
          </td>
          <td
            class="hash"
          >
            1
          </td>
        </tr>
        <tr>
          <td>
            b
          </td>
          <td
            class="hash"
          >
            123
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
